packages/@aws-cdk/aws-codepipeline-actions/lib/s3/deploy-action.ts

```diff
@@ -62,7 +77,135 @@ export class S3DeployAction extends Action {
         BucketName: this.props.bucket.bucketName,
         Extract: this.props.extract === false ? 'false' : 'true',
         ObjectKey: this.props.objectKey,
+        CannedACL: this.props.accessControl,
+        CacheControl: this.props.cacheControl && serializeCacheControlOptions(this.props.cacheControl),
       },
     };
   }
 }
+
+/**
+ * Options used to generate RFC 2616 compliant HTTP Cache-Control response header values
+ *
+ * @see https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9 RFC 2616 14.9 Cache-Control
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Cache_response_directives MDN: Cache-Control
+ */
+export interface CacheControlOptions {
+  /**
+   * Specifies the maximum amount of time a resource will be considered fresh
+   */
+  readonly maxAge?: Duration;
+
+  /**
+   * Forces caches to submit the request to the origin server for validation before releasing a cached copy
+   */
+  readonly noCache?: true;
+
+  /**
+   * The cache should not store anything about the client request or server response
+   */
+  readonly noStore?: true;
+
+  /**
+   * No transformations or conversions should be made to the resource
+   * The Content-Encoding, Content-Range, Content-Type headers must not be modified by a proxy.
+   */
+  readonly noTransform?: true;
+
+  /**
+   * Indicates that once a resource has become stale (e.g. max-age has expired),
+   * a cache must not use the response to satisfy subsequent requests
+   * for this resource without successful validation on the origin server.
+   */
+  readonly mustRevalidate?: true;
+
+  /**
+   * Indicates that the response may be cached by any cache,
+   * even if the response would normally be non-cacheable
+   */
+  readonly public?: true;
+
+  /**
+   * Indicates that the response is intended for a single user and must not be stored by a shared cache
+   * A private cache may store the response.
+   */
+  readonly private?: true;
```

Comment:
Does it ever make sense to have `{ public: true, private: true }`? Because if not, these 2 should be folded into one property with a type of an enum with 2 possible values, `PUBLIC` and `PRIVATE`.

Reply:
Grouping them into a `storageAccess` property would be counter-intuitive for a user that knows the RFC, forcing them to look into the CDK documentation. I don't think that trade-off is worth the cleaner API.

Reply:
Uhm, why? That's literally one of the tenets of the CDK, to create APIs that make it harder to shoot yourself in the foot...

Also, using properties with names like `private` and `public` will be problematic in non-JavaScript languages like Java and C#, so that's another argument for the enum approach.